specs/segragate_object_specific_flows.adoc

```diff
@@ -0,0 +1,278 @@
+= Segregate object specific flows
+
+
+All the object specific actions must be segregated under the object's namespace
+as flows. It must be noted that such a flow defined under object's namespace
+should not have dependency on any other objects' namespace. By doing this, it
+will be very easy for API to know about the list of actions that can be
+performed on the object.
+
+
+== Problem description
+
+Currently some of the actions specific to objects are defined as flows in
+global namespace of the particular entity. This does not give a clear picture
+about the object to which the flow is related.
+
+By defining the objects specific actions as flows under objects' namespace,
+we can easily relate the flows with the corresponding objects'. It will be easy
+for the API to figure out the actions that can be performed on the object. Care
+must be taken to see that such flows are not dependent on any other objects'.
+
+So in all the yaml definitions that are already available, object specific
+flows have to be moved under the specific object. This has to be considered
+while adding new yaml definitions for flows in the future as well.
+
+== Use Cases
+
+* This specification has to be followed while adding new definition to yaml
+in the future- Developer.
+
+* API layer can figure out the list of actions specific to an object by looking
+at the list of flows under that particular object- API Developer.
+
```

Comment:
what about flows which referes multiple objects. It should be as it is now(in global namespece). Please state that explicitly in the document